bin/check_versions

```diff
@@ -0,0 +1,18 @@
+#!/bin/sh
+
+need_version=0
+
+for json_file in $(git diff --name-only master HEAD | grep 'exercises/.*json' | sort | uniq); do
```

Comment:
Perhaps `grep -E '^exercises/.*\.json$'` to avoid conflicts with future exercise names that contain the word "json".

There's also `sort -u`.

Reply:
would you believe it, I completely missed `-u` when reading the manual. how did that happen...

so, is there a need for the `-E` in the proposed `grep` invocation? On the various compuers I have, `echo "exercises/a/test.json\nexercises/a/test.json.backup\nexercises/a-json-exercise/readme.md" | grep '^exercises/.*\.json$'` does the right thing (only `exercises/a/test.json`). Is there a different implementation of `grep` where the `-E` would be required, some other reason we'd want to include it, etc.?